recipe/patches/0009-add-GRPC-GRPCXX-UBP-_DLL-mechanism-for-missing-symbo.patch

```diff
@@ -0,0 +1,278 @@
++//       gRPC_DLL_IMPORTS is set by us as part of the interface for consumers of the DLL
++#if defined(gRPC_DLL_EXPORTS)
++#define GRPC_DLL __declspec(dllexport)
++#elif defined(gRPC_DLL_IMPORTS)
++#define GRPC_DLL __declspec(dllimport)
++#else
++#define GRPC_DLL
++#endif // defined(gRPC_DLL_EXPORTS)
++
++// same for gRPC++
++#if defined(gRPCXX_DLL_EXPORTS)
++#define GRPCXX_DLL __declspec(dllexport)
++#elif defined(gRPCXX_DLL_IMPORTS)
++#define GRPCXX_DLL __declspec(dllimport)
++#else
++#define GRPCXX_DLL
++#endif // defined(gRPCXX_DLL_EXPORTS)
++
+ #endif /* defined(_WIN64) || defined(WIN64) || defined(_WIN32) || \
+           defined(WIN32) */
+ 
+diff --git a/src/core/ext/upb-generated/src/proto/grpc/health/v1/health.upb.c b/src/core/ext/upb-generated/src/proto/grpc/health/v1/health.upb.c
+index 6869668ef4..ea61d71963 100644
+--- a/src/core/ext/upb-generated/src/proto/grpc/health/v1/health.upb.c
++++ b/src/core/ext/upb-generated/src/proto/grpc/health/v1/health.upb.c
+@@ -16,7 +16,7 @@ static const upb_MiniTable_Field grpc_health_v1_HealthCheckRequest__fields[1] =
+   {1, UPB_SIZE(0, 0), UPB_SIZE(0, 0), kUpb_NoSub, 9, kUpb_FieldMode_Scalar | (kUpb_FieldRep_StringView << kUpb_FieldRep_Shift)},
+ };
+ 
+-const upb_MiniTable grpc_health_v1_HealthCheckRequest_msginit = {
++const GRPCXX_DLL upb_MiniTable grpc_health_v1_HealthCheckRequest_msginit = {
+   NULL,
+   &grpc_health_v1_HealthCheckRequest__fields[0],
+   UPB_SIZE(8, 24), 1, kUpb_ExtMode_NonExtendable, 1, 255, 0,
+@@ -26,7 +26,7 @@ static const upb_MiniTable_Field grpc_health_v1_HealthCheckResponse__fields[1] =
+   {1, UPB_SIZE(0, 0), UPB_SIZE(0, 0), kUpb_NoSub, 5, kUpb_FieldMode_Scalar | (kUpb_FieldRep_4Byte << kUpb_FieldRep_Shift)},
+ };
+ 
+-const upb_MiniTable grpc_health_v1_HealthCheckResponse_msginit = {
++const GRPCXX_DLL upb_MiniTable grpc_health_v1_HealthCheckResponse_msginit = {
+   NULL,
+   &grpc_health_v1_HealthCheckResponse__fields[0],
+   UPB_SIZE(4, 8), 1, kUpb_ExtMode_NonExtendable, 1, 255, 0,
+diff --git a/src/core/ext/upb-generated/src/proto/grpc/health/v1/health.upb.h b/src/core/ext/upb-generated/src/proto/grpc/health/v1/health.upb.h
+index cff014e479..fe9fa958ea 100644
+--- a/src/core/ext/upb-generated/src/proto/grpc/health/v1/health.upb.h
++++ b/src/core/ext/upb-generated/src/proto/grpc/health/v1/health.upb.h
+@@ -24,8 +24,8 @@ struct grpc_health_v1_HealthCheckRequest;
+ struct grpc_health_v1_HealthCheckResponse;
+ typedef struct grpc_health_v1_HealthCheckRequest grpc_health_v1_HealthCheckRequest;
+ typedef struct grpc_health_v1_HealthCheckResponse grpc_health_v1_HealthCheckResponse;
+-extern const upb_MiniTable grpc_health_v1_HealthCheckRequest_msginit;
+-extern const upb_MiniTable grpc_health_v1_HealthCheckResponse_msginit;
++extern const GRPCXX_DLL upb_MiniTable grpc_health_v1_HealthCheckRequest_msginit;
++extern const GRPCXX_DLL upb_MiniTable grpc_health_v1_HealthCheckResponse_msginit;
+ 
```

Comment:
> From a cursory glance at the code, things like https://github.com/grpc/grpc/blob/27509c345cb8a4895bf3a93cb4014d012d4a0225/src/core/ext/upb-generated/src/proto/grpc/health/v1/health.upb.h#L27-L28 needs a GRPC_DLL or similar macro.

Thanks for the input. That's what I did already (even separated between grpc & grpc++, because the latter depends on the former target), and it still shows up as missing.